src/Tasks/FileIO/GetFileHash.cs

```diff
@@ -67,25 +75,27 @@ public override bool Execute()
                 return false;
             }
 
-            foreach (var file in Files)
+            Parallel.For(0, Files.Length, index =>
```

Comment:
Why not merge the two for loops? Stop the loop and task and error on first file that's not found.

Reply:
You'd waste computational resources calculating hashes, setting metadata, etc. for all the files that didn't fail. It's also messy and error-prone to stop a parallel process in the middle and make all the other parallel processes stop at the same time.

On the other hand, that would reduce startup costs. Might be an opportunity to get some telemetry and see which is better.

Reply:
I'd intended to keep the original behavior as-is, but it shouldn't be too hard to combine these. Shouldn't be incredibly difficult to compare how the two perform when everything is present, versus when some inputs are missing

Reply:
The common case is to not fail, so the potential perf gains of merging the loops would outweigh the perf hit during failures.